database/dtb.ksy

```diff
@@ -0,0 +1,128 @@
+meta:
+  id: dtb
+  title: Flattened Devicetree Format
+  file-extension: dtb
+  application:
+    - Linux
+    - Das U-Boot
+  xref:
+    - wikidata: Q16960371
+  license: CC0-1.0
+  endian: be
+doc: |
+  Also referred to as Devicetree Blob (DTB). It is a flat
+  binary encoding of data (primarily devicetree data, although
+  other data is possible as well).
+
+  On Linux systems that support this the blobs can be accessed in
+  /sys/firmware/fdt
+
+  https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-firmware-ofw
+
+  The encoding of strings used in the strings block and struct block is
+  actually a subset of ASCII:
+
+  https://github.com/devicetree-org/devicetree-specification/blob/master/source/chapter2-devicetree-basics.rst
+
+  Example files:
+
+  - https://github.com/qemu/qemu/tree/master/pc-bios
+doc-ref:
+  - https://github.com/devicetree-org/devicetree-specification/releases/tag/v0.3
+  - https://github.com/devicetree-org/devicetree-specification/blob/ba2aa679679fc4fedf67130f18a6f0ecc4cf0382/source/flattened-format.rst
+  - https://elinux.org/images/f/f4/Elc2013_Fernandes.pdf
+seq:
+  - id: magic
+    contents: [0xd0, 0x0d, 0xfe, 0xed]
+    -orig-id: magic
+  - id: total_size
+    type: u4
+    -orig-id: totalsize
```

Comment:
The key `-orig-id` should go right below `id`, see https://doc.kaitai.io/ksy_style_guide.html#seq-attr:

> ## 5. Sequence attributes
>
> When specifying an attribute, one MUST use the following order of keys:
>
> 1. Identifier(s)
>    * `id`
>    * `-orig-id` — use to specify original ID spelling if transcribing a structure from existing software and/or official spec
> 1. `size`
> 1. `size-eos`
> 1. (...)